### ISSUE TITLE
Not able to install GD in Ubuntu 17.10

### ISSUE THREAD

**Issue by vistiyos** [phpbrew/phpbrew#924]
### Output

```Downloading https://pecl.php.net/rest/r/gd/stable.txt via curl extension
CurlKit\CurlException: (22) The requested URL returned error: 404 Not Found at [https://pecl.php.net/rest/r/gd/stable.txt:The requested URL returned error: 404 Not Found]
Thrown from phar:///usr/bin/phpbrew/vendor/corneltek/curlkit/src/CurlKit/CurlDownloader.php at line 153:

  150        $ch = $this->createCurlResource($options);
  151        $data = curl_exec($ch);
  152        if (!$data) {
> 153            throw new CurlException($ch, $url . ":" . curl_error($ch) );
  154        }
  155
  156        // We don't enable CURLOPT_FOLLOWLOCATION because
  157        // the progress bar does not work after the 1st redirect..

Trace:

    0) CurlKit\CurlDownloader->request('https://pecl.php.net/rest/r/gd/stable.txt')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Downloader/PhpCurlDownloader.php: 39

    1) PhpBrew\Downloader\PhpCurlDownloader->process('https://pecl.php.net/rest/r/gd/stable.txt', '/tmp/phpbrew_Y1E0ZE')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Downloader/BaseDownloader.php: 49

    2) PhpBrew\Downloader\BaseDownloader->download('https://pecl.php.net/rest/r/gd/stable.txt')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Downloader/BaseDownloader.php: 69

    3) PhpBrew\Downloader\BaseDownloader->request('https://pecl.php.net/rest/r/gd/stable.txt')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Extension/Provider/PeclProvider.php: 47

    4) PhpBrew\Extension\Provider\PeclProvider->getPackageXml('gd', 'stable')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Extension/Provider/PeclProvider.php: 74

    5) PhpBrew\Extension\Provider\PeclProvider->buildPackageDownloadUrl('stable')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Extension/ExtensionDownloader.php: 35

    6) PhpBrew\Extension\ExtensionDownloader->download(PhpBrew\Extension\Provider\PeclProvider, 'stable')
        from phar:///usr/bin/phpbrew/src/PhpBrew/Command/ExtensionCommand/InstallCommand.php: 156

    7) PhpBrew\Command\ExtensionCommand\InstallCommand->execute('gd', '--', '--with-libdir=lib/x86_64-linux-gnu', '--with-gd=shared', '--enable-gd-native-ttf', '--with-jpeg-dir=/usr', '--with-png-dir=/usr', '--with-freetype-dir=/usr/include/freetype2/ft2build.h', '--with-xpm-dir=/usr')
PHP Notice:  Undefined index: file in phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php on line 101
PHP Stack trace:
PHP   1. {main}() /usr/bin/phpbrew:0
PHP   2. require() /usr/bin/phpbrew:99
PHP   3. PhpBrew\Console->runWithTry() phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php:2
PHP   4. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dump() phar:///usr/bin/phpbrew/src/PhpBrew/Console.php:157
PHP   5. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dumpTrace() phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php:148

Notice: Undefined index: file in phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php on line 101

Call Stack:
    0.0004     398160   1. {main}() /usr/bin/phpbrew:0
    0.0166     889024   2. require('phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php') /usr/bin/phpbrew:99
    0.0279    1417224   3. PhpBrew\Console->runWithTry() phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php:2
    1.1091    4610816   4. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dump() phar:///usr/bin/phpbrew/src/PhpBrew/Console.php:157
    1.1099    4610816   5. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dumpTrace() phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php:148

PHP Notice:  Undefined index: line in phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php on line 101
PHP Stack trace:
PHP   1. {main}() /usr/bin/phpbrew:0
PHP   2. require() /usr/bin/phpbrew:99
PHP   3. PhpBrew\Console->runWithTry() phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php:2
PHP   4. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dump() phar:///usr/bin/phpbrew/src/PhpBrew/Console.php:157
PHP   5. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dumpTrace() phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php:148

Notice: Undefined index: line in phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php on line 101

Call Stack:
    0.0004     398160   1. {main}() /usr/bin/phpbrew:0
    0.0166     889024   2. require('phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php') /usr/bin/phpbrew:99
    0.0279    1417224   3. PhpBrew\Console->runWithTry() phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php:2
    1.1091    4610816   4. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dump() phar:///usr/bin/phpbrew/src/PhpBrew/Console.php:157
    1.1099    4610816   5. CLIFramework\ExceptionPrinter\DevelopmentExceptionPrinter->dumpTrace() phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/ExceptionPrinter/DevelopmentExceptionPrinter.php:148

        from : 0

    8) call_user_func_array([PhpBrew\Command\ExtensionCommand\InstallCommand, 'execute'], ['gd', '--', '--with-libdir=lib/x86_64-linux-gnu', '--with-gd=shared', '--enable-gd-native-ttf', '--with-jpeg-dir=/usr', '--with-png-dir=/usr', '--with-freetype-dir=/usr/include/freetype2/ft2build.h', '--with-xpm-dir=/usr'])
        from phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/CommandBase.php: 845

    9) CLIFramework\CommandBase->executeWrapper(['gd', '--', '--with-libdir=lib/x86_64-linux-gnu', '--with-gd=shared', '--enable-gd-native-ttf', '--with-jpeg-dir=/usr', '--with-png-dir=/usr', '--with-freetype-dir=/usr/include/freetype2/ft2build.h', '--with-xpm-dir=/usr'])
        from phar:///usr/bin/phpbrew/vendor/corneltek/cliframework/src/Application.php: 398

    10) CLIFramework\Application->run(['/usr/bin/phpbrew', '-d', 'ext', 'install', 'gd', '--', '--with-libdir=lib/x86_64-linux-gnu', '--with-gd=shared', '--enable-gd-native-ttf', '--with-jpeg-dir=/usr', '--with-png-dir=/usr', '--with-freetype-dir=/usr/include/freetype2/ft2build.h', '--with-xpm-dir=/usr'])
        from phar:///usr/bin/phpbrew/src/PhpBrew/Console.php: 119

    11) PhpBrew\Console->runWithTry(['/usr/bin/phpbrew', '-d', 'ext', 'install', 'gd', '--', '--with-libdir=lib/x86_64-linux-gnu', '--with-gd=shared', '--enable-gd-native-ttf', '--with-jpeg-dir=/usr', '--with-png-dir=/usr', '--with-freetype-dir=/usr/include/freetype2/ft2build.h', '--with-xpm-dir=/usr'])
        from phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php: 2

    12) require('phar:///usr/bin/phpbrew/scripts/phpbrew-emb.php')
        from /usr/bin/phpbrew: 99

...
```

### Expected Result

```
Get GD installed. 
```

### Command

phpbrew -d ext install gd -- --with-libdir=lib/x86_64-linux-gnu --with-gd=shared --enable-gd-native-ttf --with-jpeg-dir=/usr --with-png-dir=/usr --with-freetype-dir=/usr/include/freetype2/ft2build.h

### Platform

OS: Ubuntu 17.10

Running PHP: PHP 7.1.11

Installing PHP: PHP 7.1.11

**Comment by morozov** [comment]
@vistiyos most probably, PHPBrew cannot find the PHP sources to build GD and thinks it's a PECL extension. Please make sure that the sources are in place (`~/.phpbrew/build/php-7.1.11/ext/gd`).

If not, you can try running `phpbrew install --dryrun 7.1.11` so that PHPBrew re-downloads and unpacks them.

**Comment by vistiyos** [comment]
Now I'm getting this:
```
make: *** No rule to make target '/home/victor/.phpbrew/build/php-7.1.11/ext/gd/gd.c', needed by 'ext/gd/gd.lo'.  Stop.

**Comment by morozov** [comment]
@vistiyos it looks like having the source in place is not enough. I was able to reproduce the same error by doing this:
```
↪ cd .phpbrew/build/php-7.1.11
↪ make distclean
↪ phpbrew ext install gd
```
Given that your build files indeed were (at least partially) erased, I'd suggest re-installing the build from scratch and then installing/enabling `gd`.

**Comment by sam452** [comment]
I'm having this same issue on 7.0.0 (unable to update and find 7.1.1) for Ubuntu jessie. Efforts to install the gd extension fails with the same error. Where should it find this binary?

**Comment by morozov** [comment]
@sam452 please try removing and reinstalling you PHP. Installation of `gd` is not supposed to download anything from PECL. It's a sign of PHPBrew not being able to find the bundled `gd` extension and therefore thinking it's a PECL extension.

**Comment by edigu** [comment]
I am trying to compile php using gd with freetype support on macos since two days with no success. I have shared my experience here: https://github.com/phpbrew/phpbrew/issues/779 and it looks like there is an issue. `phpbrew ext install gd -- --options` immediately going to PECL rest API and resulting with a 404, then exception. I tried to install with `+gd` as well but phpbrew compiled it without freetype.
